### PR TITLE
VZ:10047 update security context of fluentbit container

### PR DESCRIPTION
**Pull request by karan56625**
Add container level security context to fluentbit container.

### CODE REVIEW

File: platform-operator/helm_config/overrides/fluent-operator-values.yaml [hunk 1]
```diff
@@ -54,6 +54,12 @@ fluentbit:
   podSecurityContext:
     seccompProfile:
       type: RuntimeDefault
+  securityContext:
+    allowPrivilegeEscalation: false
+    privileged: false
+    capabilities:
+      drop:
+      - ALL
```
Comment (adiforluls) [inline platform-operator/helm_config/overrides/fluent-operator-values.yaml:62]: Simply dropping all capabilities works right (i.e. you don't have to add any capability back)?

**Review by adiforluls — APPROVED** [review]

File: platform-operator/helm_config/overrides/fluent-operator-values.yaml (hunk 1 above)
Comment (karan56625) [reply platform-operator/helm_config/overrides/fluent-operator-values.yaml:62]: yes

Comment (pfmackin) [reply platform-operator/helm_config/overrides/fluent-operator-values.yaml:62]: can you add 
    runAsGroup: 1025
    runAsNonRoot: true
    runAsUser: 1025

File: platform-operator/thirdparty/charts/fluent-operator/values.yaml [hunk 2]
```diff
@@ -85,6 +85,8 @@ fluentbit:
   secrets: []
   # Pod security context for Fluent Bit pods. Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
   podSecurityContext: {}
+  # Security context for Fluent Bit container. Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  securityContext: {}
```
Comment (pfmackin) [inline platform-operator/thirdparty/charts/fluent-operator/values.yaml:89]: So for fluentbit itself there is no securityContext?

Comment (karan56625) [reply platform-operator/thirdparty/charts/fluent-operator/values.yaml:89]: We have added support to add securityContext for fluentbit in upstream. We have cherry-picked that here. So we are adding fluentbit Securitycontext

File: platform-operator/helm_config/overrides/fluent-operator-values.yaml (hunk 1 above)
Comment (karan56625) [reply platform-operator/helm_config/overrides/fluent-operator-values.yaml:62]: fluentbit is having issue while running as non-root user. We have tried running it as non root and providing some capabilities, but that's not enough. so running it as root user and dropping all capabilities.

Comment (karan56625) [reply platform-operator/helm_config/overrides/fluent-operator-values.yaml:62]: will raise follow up PR if we able to run the fluentbit as non-root user.

**Pull request merged**